cbits/ctermid.c

```diff
@@ -0,0 +1,21 @@
+/* Provide ctermid() if it is missing (as on some Android versions). */
+
+#include "HsUnixConfig.h"
+#ifndef HAVE_CTERMID
+
+#include "ghcplatform.h"
+
+#if defined(linux_HOST_OS) || defined(linux_android_HOST_OS)
```

Comment:
why are we testing for a specific OS, rather than follow Autoconf's idiom of testing for features?